webapp/sources/utils/src/main/scala/zio/internal/RudderExecutor.scala

```diff
@@ -39,11 +41,37 @@ trait RudderBlockingService extends Blocking {
       )
 
       threadPool
-    })
+    }
 
     // this doesn't yield better results
     // val scalaGobal: UIO[Executor] = ZIO.succeed(Executor.fromExecutionContext(Int.MaxValue)(scala.concurrent.ExecutionContext.global))
 
-    val blockingExecutor: UIO[Executor] = zio
+    override val blockingExecutor: Executor = zio
   }
 }
+
+
+object RudderAsyncExecutor {
+
+final def makeDefault(yieldOpCount: Int): Executor =
+    Executor.fromThreadPoolExecutor(_ => yieldOpCount) {
+      val corePoolSize  = Runtime.getRuntime.availableProcessors() * 2
```

Comment:
why twice the number of core ? is it documented somewhere the reasoning ?

Reply:
Actually, we don't use it at all anymore. I think it was just for tests. 
The reasoning was: we have a lot of threads used to process blocking fiber, and most of them actually block. So it is reasonnable to have a core pool size of CPUx2, given most of them will be block often. 
In practice, IIRC, it worked well for small number of CPU, but for more (like 16), it was a bit too big.